src/main/java/me/magnet/consultant/Consultant.java

```diff
@@ -665,11 +666,17 @@ private void updateValidatedConfig(Properties newConfig) {
 	 * @throws InterruptedException If it got interrupted while waiting for any open tasks
 	 */
 	public void shutdown() throws InterruptedException {
-		try {
-			deregisterService();
+		if (registered.get()) {
+			try {
+				deregisterService();
+			}
+			catch (ConsultantException e) {
+				log.error("Error occurred while deregistering", e);
+			}
 		}
-		catch (ConsultantException e) {
-			log.error("Error occurred while deregistering", e);
+
+		if (pullConfig && !executor.isShutdown()) {
+			MoreExecutors.shutdownAndAwaitTermination(executor, TERMINATION_TIMEOUT_SECONDS, TimeUnit.SECONDS);
```

Comment:
`shutdownAndAwaitTermination` doesn't throw, however we should probably log its return type indicating all tasks have successfully been completed. Personally I say we should log those as a warning, since at the point of shutdown, the config updates no longer affect the system, and they wouldn't leave the client in a broken state or something like that.